### PR TITLE
Fix FORTRAN type mismatch on allocate_pointer calls.

### PR DESCRIPTION
**Pull request by aaroncblack**
There was a type mismatch on the allocate_pointer calls.  UM-351.

Code was passing in an INTEGER, and interface was expecting a INTEGER(C_SIZE_T).   Some compilers, like GNU, don't enforce this, but other compilers (XL) issued an error message on the mismatch.

I updated the shroud generator script, but do not know how to produce the generated bindings.  For now, I manually updated the generated files.  Before this pull request is merged I need a team member to inform me how to regenerate the bindings and test that.

### CODE REVIEW

**Comment by aaroncblack** [comment]
@davidbeckingsale - Should this be merged?  Need an approval if so

**Comment by davidbeckingsale** [comment]
Yes, we can merge it - I was waiting for you to update the CHANGELOG.

**Comment by aaroncblack** [comment]
Oh yeah, forgot about that.  Ok, changed it.

- Aaron

File: CHANGELOG.md [hunk 1]
```diff
@@ -120,3 +120,7 @@ out new cookbooks.
 - REPLAY previously recorded some operations with multiple print statements
 causing REPLAY output from MPI runs to become mixed between multiple ranks.
 REPLAY has been modified to output each operation onto a single line.
+
+- Update calls to allocate_pointer in the FORTRAN bindings to ensure that the
```
Comment (davidbeckingsale) [inline CHANGELOG.md]: This should be in the 'Fixed' section at the top of the file (under "Unreleased")

Comment (davidbeckingsale) [reply CHANGELOG.md]: @aaroncblack are you able to make this change?

Comment (davidbeckingsale) [reply CHANGELOG.md]: ?

**Pull request merged**